### PR TITLE
feat(postgres): add a describe method

### PR DESCRIPTION
**Pull request by raphaelvignes**
## Change Summary

In order to enable a "Postgres/Redshift" VQB translator, we need to provide a describe method to the Postgres connector.
My implementation is quite naive (opening an adhoc connection) but it should do the job.

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable

### CODE REVIEW

File: toucan_connectors/postgres/postgresql_connector.py [hunk 1]
```diff
@@ -161,3 +162,10 @@ def get_status(self) -> ConnectorStatus:
                 )
 
         return ConnectorStatus(status=True, details=self._get_details(3, True), error=None)
+
+    def describe(self, data_source):
+        connection = pgsql.connect(**self.get_connection_params(database=data_source.database))
+        with connection.cursor() as cursor:
+            cursor.execute(f"""select * from ({data_source.query.replace(';','')}) as q LIMIT 0;""")
```
Comment (Sanix-Darker) [inline toucan_connectors/postgres/postgresql_connector.py:169]: I think, for everything related to sql, let keep all in uppercase like on snowflake !
'SELECT', 'FROM', ....
Even for column name, that will be easy for the future collaboration with snowflake source code i think !

File: tests/postgres/test_postgres.py [hunk 2]
```diff
@@ -284,3 +284,39 @@ def test_get_form_connection_fails(mocker, postgres_connector):
     mocker.patch.object(pgsql, 'connect').side_effect = IOError
     form = PostgresDataSource.get_form(postgres_connector, current_config={})
     assert 'table' in form['properties']
+
+
+def test_describe(mocker, postgres_connector):
+    """It should return a table description"""
+    ds = PostgresDataSource(
+        domain='test', name='test', database='postgres_db', query='select * from city;'
```
Comment (Sanix-Darker) [inline tests/postgres/test_postgres.py:292]: uppercase query 🙏🏾

File: tests/postgres/test_postgres.py [hunk 3]
```diff
@@ -284,3 +284,39 @@ def test_get_form_connection_fails(mocker, postgres_connector):
...
+        'countrycode': 'bpchar',
+        'district': 'text',
+        'population': 'int4',
+    }
+
+
+def test_describe_error(mocker, postgres_connector):
+    """It should return a table description"""
+    ds = PostgresDataSource(
+        domain='test', name='test', database='postgres_db', query='select * from city;'
```
Comment (Sanix-Darker) [inline tests/postgres/test_postgres.py:307]: uppercase query 🙏🏾

**Review by Sanix-Darker — APPROVED** [review]
Thanks for this PR !
i left some minors comments for you !

File: toucan_connectors/postgres/postgresql_connector.py (hunk 1 above)
Comment (raphaelvignes) [reply toucan_connectors/postgres/postgresql_connector.py]: thx.

File: tests/postgres/test_postgres.py (hunk 2 above)
Comment (raphaelvignes) [reply tests/postgres/test_postgres.py]: :+1:

File: tests/postgres/test_postgres.py (hunk 3 above)
Comment (raphaelvignes) [reply tests/postgres/test_postgres.py]: :+1:

**Pull request merged**